Dockerfile

```diff
@@ -109,6 +109,9 @@ RUN apt-get update \
     && rm -rf /var/lib/apt/lists/*
 
 ARG APP=/qdrant
+ARG USERNAME=root
+ARG USER_UID=0
+ARG USER_GID=0
```

Comment:
I suggest moving this close to the point of first use. (E.g., right before `RUN` that creates the user.)
It provides better caching.

Also, I'd personally reduce this to a single `USER_ID` ARG and derive everything else from there. E.g., use `USER_ID` for both `uid` and `gid` and simply "hardcode" the username to `user` or `qdrant`.

Reply:
I simplified and re-ordered it as suggested.